zfs/zfs.go

```diff
@@ -1919,3 +1919,21 @@ func ZFSRollback(ctx context.Context, fs *DatasetPath, snapshot FilesystemVersio
 
 	return err
 }
+
+func FilterCheckFS(fsfilter DatasetFilter, fs string) (*DatasetPath, error) {
```

Comment:
Can we have this in a separate PR? Maybe we can start using generics for this.

Reply:
Do you mean just break this one hunk into a separate PR, which the current one then gets dependent on? (As the meat of  this PR is inherently coupled to this addition)

If so; sure, if not; I may need some more elaboration on what's intended

Reply:
> As the meat of this PR is inherently coupled to this addition

Why is that?

Reply:
...I'll get some sleep and get back to this :face_in_clouds: